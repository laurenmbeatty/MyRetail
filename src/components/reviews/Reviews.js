import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import * as Styled from "./styles";
class Reviews extends React.Component {
  manipulateDate = info => {
    let newDate = "";
    const month = info.slice(4, 7);
    info = new Date(info);
    const year = info.getFullYear();
    const day = info.getDate();

    newDate = `${month} ${day}, ${year}`;
    return newDate;
  };

  render() {
    const { reviews } = this.props;
    const starsArray = [1, 2, 3, 4, 5];

    return (
      <Styled.ReviewSection>
        {reviews && (
          <Styled.ReviewHeading>
            <div className="ratings-stars-container">
              <div className="ratings-stars">
                {starsArray.map(num =>
                  num <= parseInt(reviews.consolidatedOverallRating) ? (
                    <FontAwesomeIcon
                      key={num}
                      icon="star"
                      className="icon-star active"
                    />
                  ) : (
                    <FontAwesomeIcon
                      key={num}
                      icon="star"
                      className="icon-star"
                    />
                  )
                )}
              </div>
              <span className="ratings-label">overall</span>
            </div>
            <a href="/" className="view-all-ratings">
              view all {reviews.totalReviews} reviews
            </a>
          </Styled.ReviewHeading>
        )}
        {reviews && (
          <Styled.ReviewMain>
            <div className="review-main-heading">
              <div className="pro-con">
                <p>PRO</p>
                <span>most helpful 4-5 star review</span>
              </div>
              <div className="pro-con">
                <p>CON</p>
                <span>most helpful 1-2 star review</span>
              </div>
            </div>
            <hr />
            <Styled.ProConReviewContainer>
              <Styled.Review>
                <div className="ratings-stars">
                  {starsArray.map(num =>
                    num <= parseInt(reviews.Pro[0].overallRating) ? (
                      <FontAwesomeIcon
                        key={num}
                        icon="star"
                        className="icon-star active"
                      />
                    ) : (
                      <FontAwesomeIcon
                        key={num}
                        icon="star"
                        className="icon-star"
                      />
                    )
                  )}
                </div>
                <h3 className="review-title">{reviews.Pro[0].title}</h3>
                <p className="review-description">{reviews.Pro[0].review}</p>
                <div className="review-footer">
                  <span className="reviewer">{reviews.Pro[0].screenName}</span>
                  <span className="review-date">
                    {this.manipulateDate(reviews.Pro[0].datePosted)}
                  </span>
                </div>
              </Styled.Review>
              <Styled.Review>
                <div className="ratings-stars">
                  {starsArray.map(num =>
                    num <= parseInt(reviews.Con[0].overallRating) ? (
                      <FontAwesomeIcon
                        key={num}
                        icon="star"
                        className="icon-star active"
                      />
                    ) : (
                      <FontAwesomeIcon
                        key={num}
                        icon="star"
                        className="icon-star"
                      />
                    )
                  )}
                </div>
                <h3 className="review-title">{reviews.Con[0].title}</h3>
                <p className="review-description">{reviews.Con[0].review}</p>
                <div className="review-footer">
                  <span className="reviewer">{reviews.Con[0].screenName}</span>
                  <span className="review-date">
                    {this.manipulateDate(reviews.Con[0].datePosted)}
                  </span>
                </div>
              </Styled.Review>
            </Styled.ProConReviewContainer>
          </Styled.ReviewMain>
        )}
      </Styled.ReviewSection>
    );
  }
}
export default Reviews;
