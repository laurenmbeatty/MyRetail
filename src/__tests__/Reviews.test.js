import React from "react";
import { create } from "react-test-renderer";
import "jest-styled-components";
import { library } from "@fortawesome/fontawesome-svg-core";
import { faStar } from "@fortawesome/free-solid-svg-icons";
library.add(faStar);
import Reviews from "../components/reviews/Reviews";

test("snapshot", () => {
  const c = create(<Reviews />);
  expect(c.toJSON()).toMatchSnapshot();
});

test("Parses date Correctly", () => {
  const c = create(<Reviews manipulateDate={() => {}} />);
  const instance = c.getInstance();
  const testDate = "Mon Mar 11 13:13:55 UTC 2013";

  expect(instance.manipulateDate(testDate)).toBe("Mar 11, 2013");
});
