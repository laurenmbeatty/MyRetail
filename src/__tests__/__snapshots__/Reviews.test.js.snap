// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<section
  className="sc-bdVaJa bDWFJH"
>
  <div
    className="sc-bwzfXH gRmwoP"
  >
    <div
      className="ratings-stars-container"
    >
      <div
        className="ratings-stars"
      />
      <span
        className="ratings-label"
      >
        overall
      </span>
    </div>
    <a
      className="view-all-ratings"
      href="/"
    >
      view all 
      14
       reviews
    </a>
  </div>
  <div
    className="sc-htpNat kaXRpo"
  >
    <div
      className="review-main-heading"
    >
      <div
        className="pro-con"
      >
        <p>
          PRO
        </p>
        <span>
          most helpful 4-5 star review
        </span>
      </div>
      <div
        className="pro-con"
      >
        <p>
          CON
        </p>
        <span>
          most helpful 1-2 star review
        </span>
      </div>
    </div>
    <hr />
    <div
      className="sc-bxivhb gSSgMV"
    >
      <div
        className="sc-ifAKCX fenHHe"
      >
        <div
          className="ratings-stars"
        />
        <h3
          className="review-title"
        >
          Fantastic Blender
        </h3>
        <p
          className="review-description"
        >
          This blender works amazingly, and blends within seconds.  The single serve cups also work really well for smoothies or protein shakes!
        </p>
        <div
          className="review-footer"
        >
          <span
            className="reviewer"
          >
            Eric
          </span>
          <span
            className="review-date"
          >
            Apr 18, 2013
          </span>
        </div>
      </div>
      <div
        className="sc-ifAKCX fenHHe"
      >
        <div
          className="ratings-stars"
        />
        <h3
          className="review-title"
        >
          Very unhappy
        </h3>
        <p
          className="review-description"
        >
          Less than 2 months after purchase it completely stopped working. First it wouldn't detect the pitcher when trying to blend a significant amount, a couple weeks later it wouldn't detect the single serve cup. 
        </p>
        <div
          className="review-footer"
        >
          <span
            className="reviewer"
          >
            New York
          </span>
          <span
            className="review-date"
          >
            Mar 11, 2013
          </span>
        </div>
      </div>
    </div>
  </div>
</section>
`;
